compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -3520,6 +3520,8 @@ pub fn trimmed_def_paths(tcx: TyCtxt<'_>, (): ()) -> DefIdMap<Symbol> {
 
     // Put the symbol from all the unique namespace+symbol pairs into `map`.
     let mut map: DefIdMap<Symbol> = Default::default();
+    // Precatious is taken below in the loop so we can allow this lint here.
```

Comment:
```suggestion
    // Precautions are taken below in the loop so we can allow this lint here.
```

Reply:
Tho this is only used in diagnostics, right? we could just use an IndexMap for `unique_symbols_rev`, too?

Reply:
Done.